src/components/ProductDetail.jsx

```diff
@@ -0,0 +1,48 @@
+import Loading from './../components/Loading';
+import { useFetch } from './../JS/useFetch';
+import { useParams, useNavigate } from 'react-router-dom';
+
+
+const ProductDetail = () => {
+    const productID = Number(useParams().productID) || 0;
```

Comment:
That's one way to do it, but I'd recommend destructuring the `useParams()` hook.

`const { productId } = Number(useParams()) || 0`

Mainly because you could always use the pattern even if you were passing multiple values

```
// current implementation
    const productID = Number(useParams().productID) || 0;
    const type = Number(useParams().type) || '';
// destructuring implementation
    const { productID, type } = Number(useParams());
    // then handle any default values you want to enforce.

```